src/Repository/CreditRepository.php

```diff
@@ -0,0 +1,131 @@
+<?php
+namespace App\Repository;
+
+use Wikimedia\ToolforgeBundle\Service\ReplicasClient;
+
+class CreditRepository {
+	/** @var ReplicasClient */
+	private $client;
+
+	public function __construct( ReplicasClient $client ) {
+		$this->client = $client;
+	}
+
+	public function getDatabaseName( $domain, $family ) {
+		if ( $family === 'wikisource' ) {
+			if ( in_array( $domain, [ 'old', 'www', 'mul' ] ) ) {
+				$dbname = 'sourceswiki';
+			} elseif ( $domain === 'zh-min-nan' ) {
+				$dbname = 'zh_min_nanwikisource';
+			} else {
+				$dbname = $domain . $family;
+			}
+		} else {
+			$dbname = $domain . $family;
+		}
+		return $dbname;
+	}
+
+	public function getPageCredits( $lang, $namespaces, $pages ) : array {
+		$dbname = $this->getDatabaseName( $lang, 'wikisource' );
+
+		$connection = $this->client->getConnection( $dbname );
+		$pageNames = [];
+		$namespaceName = "";
+		$splittedPages = [];
+
+		foreach ( $pages as $page ) {
+			$splitPage = explode( ":", $page, 2 );
+			if ( isset( $splitPage[1] ) ) {
+				$namespaceName = $splitPage[0];
+				$pageNames[] = $splitPage[1];
+				if ( isset( $namespaces[$namespaceName] ) ) {
```

Comment:
Agree. I followed your logic and it's correct. I just have a minor question. For the cases like `Foo:Bar`, do we want to assume that the page might be in the main namespace or should we omit it completely since `Foo` is not a valid namespace?

Reply:
Any page can have a colon in the title, so no, they should never be skipped. A real-worl example is [John Huss: his life, teachings and death, after five hundred years](https://en.wikisource.org/wiki/John_Huss:_his_life,_teachings_and_death,_after_five_hundred_years). Obviously `John Huss` is not a namespace, hence the full page title (before and after the colon) is a mainspace title.

Reply:
Got it. I did not know that a page title's can contain a color. Thanks!